applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -1039,6 +1164,46 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::SelectUpwindEle
     }
 }
 
+template<int TDim, int TNumNodes>
+int TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::GetAdditionalNode() const
+{
+    const TransonicPerturbationPotentialFlowElement& r_this = *this;
+
+    // current and upwind element geometry
+    const GeometryType& r_geom = r_this.GetGeometry();
+    const GeometryType& r_upwind_geom = mpUpwindElement->GetGeometry();
+
+    int non_matching_nodes_counter = 0;
+    int additional_node = -1;
+    bool loop_stop = false;
+
+    for (unsigned int i = 0; i < TNumNodes && !loop_stop; i++)
+    {
+        for (unsigned int j = 0; j < TNumNodes && !loop_stop; j++)
+        {
+            // check if nodes are the same
+            if(r_upwind_geom[i].Id() != r_geom[j].Id())
+            {
+                non_matching_nodes_counter++;
+            }
+        }
+        // get index of non matching node
+        if (non_matching_nodes_counter == TNumNodes)
+        {
+            additional_node = i;
+            loop_stop = true;
+        }
+        non_matching_nodes_counter = 0;
+    }
+
+    if(additional_node == -1)
+    {
+        KRATOS_WARNING("GetAdditionalNode") << " did not find additional node for element # " << r_this.Id() << std::endl;
+    }
+
```

Comment:
I really like the warning here, makes total sense! I find it brings robustness and security to your implementation. I would actually even throw and error if no additional node is found, it might save you some headaches looking for a bug later on.
```suggestion
    KRATOS_ERROR_IF(additional_node == -1) << "No upwind element id found for element #" << this->Id() << std::endl;

```